### ISSUE TITLE
The book should document declarative defaulting

### ISSUE THREAD

**Issue by DirectXMan12** [kubernetes-sigs/kubebuilder#1622]
We should document declarative defaulting better in the book, since it's been supported for quite a few versions now.  We'll need to warn about Kubernetes versions with a note like we do here: https://github.com/kubernetes-sigs/kubebuilder/blame/master/docs/book/src/multiversion-tutorial/tutorial.md#L26, but we should reccomend this over using mutating webhooks when possible.

/kind feature

**Comment by DirectXMan12** [comment]
/good-first-issue

(this entails moving the current "defaulting webhook" documentation to a standalone tutorial and editing the page where we go over types and validation markers to talk about defaulting markers too)

**Comment by stijndehaes** [comment]
I see you are assigned @krol3, don't know if you are working on it or this is still relevant. But if it's still relevant and you are ok with this I would love to give this a try :)

**Comment by krol3** [comment]
@stijndehaes please give a try. I didn't have time to do it.

**Comment by stijndehaes** [comment]
@DirectXMan12 Since it's been a long time ago I think it's better that I propose what I think I have to do before starting so here it goes:

- The page about implementing defaulting/validating webhook has to be moved to reference. Since we want people to use default/validation markers in the tutorial
- Add a new page in the tutorial instead of the webhook page that uses a default marker and a validation marker. Only thing I am not sure of is the `validateScheduleFormat` function this is impossible to write with validation markers I believe. So maybe for that we keep a validating webhook in the tutorial? Or we might just past a link to the reference defaulting/validating webhook page.

If my idea is ok please let me know and I'll get started on it :) Otherwise please feel free to give me some of your ideas.

**Comment by parul5sahoo** [comment]
@camilamacedo86 @DirectXMan12 , Can I work on this issue?

**Comment by camilamacedo86** [comment]
Yes, it is not assigned to anybody. 
Please, feel free to get this one.

**Comment by parul5sahoo** [comment]
Hey @camilamacedo86 having gone through the previous comments I suppose, I need to move [this document](https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation.html) into a tutorial in [this section](https://book.kubebuilder.io/multiversion-tutorial/webhooks.html#setting-up-the-webhooks) as 2.3.2 or 2.3.1.1... where we take this into consideration : 
> but we should recommend this over using mutating webhooks when possible.

Then add a new page in [this section](https://book.kubebuilder.io/reference/markers.html)  namely 5.6.7 to talk about default markers as mentioned in 

> (this entails moving the current "defaulting webhook" documentation to a standalone tutorial and editing the page where we go over types and validation markers to talk about defaulting markers too)

**Comment by camilamacedo86** [comment]
Hi @parul5sahoo,

Hello, are you working on this one? If not could you please unsigned so that you let others try to check this one?

**Comment by ashutosh887** [comment]
Hey @sayantani11 if you wish we can work on this issue together
/assign

**Comment by camilamacedo86** [comment]
Hi @sayantani11 and @ashutosh887,

Are you working on this one? If not, could you please remove the /assign?

**Comment by mogsie** [comment]
I suggest a new section be added in front of the validating webhooks that simply implements the same defaulting as the validating webhooks do now.  They're probably more than capable of implementing it.  What to do with the webhooks?  Maybe there is some more complicated validation that could be done? 

* Maybe that you can't change the concurrency policy from Allow to Forbid if there is an active job?
* Or that the default varies depending on the schedule?  e.g. if you have something that runs per minute, then more aggressive defaults (or limits) could be imposed, so you aren't allowed to create 1440 jobs per day.  Something advanced that warrants the webhook approach.

**Comment by camilamacedo86** [comment]
With the initiatives to change the current implementation to use the new CustomInterfaces those examples have been improved as well. We still needing to address changes on the master branch regards those breaking changes in the Controller Runtime to decouple the webhooks from the APIs.

So, I am closing this one for now, since it is very old and improvements are addressed in this area.
However,  after the next release 4.3.0 if someone see that more info still required and can be added in the docs please feel free to raise a new issue and/or an PR. 

Thank you for all your collab 🥇
